.buildkite/scripts/e2e/bootstrap.py

```diff
@@ -0,0 +1,177 @@
+"""
+E2E bootstrapping with Python script
+    - A script to resolve stack versions
+    - Download and spin up elastic-package
+    - Clone integrations repo and prepare packages
+    - When E2E finishes, teardown the stack
+"""
+import io
+import os
+import subprocess
+import tarfile
+import time
+from logstash_stats import LogstashStats
+from util import Util
+
+
+class Bootstrap:
+    ELASTIC_PACKAGE_DISTRO_URL = "https://api.github.com/repos/elastic/elastic-package/releases/latest"
+    LOGSTASH_CONTAINER_NAME = "elastic-package-stack-e2e-logstash-1"
+    PLUGIN_NAME = "logstash-filter-elastic_integration"
+
+    logstash_stats_api = LogstashStats()
+
+    def __init__(self, stack_version, platform):
+        f"""
+        A constructor of the {Bootstrap}.
+
+        Args:
+            stack_version (str): An Elastic stack version where {Bootstrap} spins up with
+            platform (str): pass macos if you are on your dev Mac, otherwise linux 
+
+        Returns:
+            void: validates platform, sets stack version and platform
+        """
+        self.stack_version = stack_version
+        self.__validate_platform(platform)
+        self.__set_distro(platform)
+
+    def __validate_platform(self, platform):
+        platforms = ["macos", "linux"]
+        if platform not in platforms:
+            raise ValueError(f"platform accepts {platforms}")
+        self.platform = platform
+
+    def __set_distro(self, platform):
+        self.distro = "darwin_amd64.tar.gz" if platform == "macos" else "linux_amd64.tar.gz"
+
+    def __download_elastic_package(self):
+        response = Util.call_url_with_retry(self.ELASTIC_PACKAGE_DISTRO_URL)
+        release_info = response.json()
+
+        download_urls = [asset["browser_download_url"] for asset in release_info["assets"]]
+        download_url = [url for url in download_urls if self.distro in url][0]
+        if download_url.strip() == '':
+            raise Exception("Could not resolve elastic-package distro download URL.")
+
+        os.system(f"wget {download_url}")
+        print("elastic-package is successfully downloaded.")
+
+        # Extract the downloaded tar.gz file
+        with tarfile.open(download_url.split("/")[-1], "r:gz") as tar:
+            tar.extractall()
+
+    def __make_elastic_package_global(self):
+        result = subprocess.run(["sudo", "mv", "elastic-package", "/usr/local/bin"])
```

Comment:
Correct me if I am wrong: my 5-6 yrs old memory says non using shell=True is the best practise because it will be OS dependant and shell executions injection processes may fail regardless of current task.
I couldn't find proper blog/guide what changed recently and it is still same. ChatGPT says it is still not recommended 😆😁

Reply:
Good questions.
Re: os dependency, do you we plan to run this on >1 different OSes?
The reason I suggested to use `shell=True`, at times, is a) in case you want to bundle things like pipes into your commands b) rely on preexisting env vars.

If you have tested that everything works fine without `shell=True`, e.g. starting up `elastic-package` feel free to ignore the comment and leave it as is.

Reply:
I have once more went through `subprocess` package and do see it is pretty well abstracted. However, with _os-dependant_ `Shell=True` I may break something especially when testing on local (mac) and then push to our BK agents. If it is _strong_ suggestion I am open to change and I do believe you have stronger experience on these kinda cases than me.